Sources/ClientRuntime/Identity/IdentityAPI/Identity.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Base protocol for all identity types
+public protocol Identity {
+    var expiration: Date? { get }
+}
+
+// Identity v. IdentityT v. IdentityKind
+// - Identity is the protocol that all identity types must conform to.
+// - IdentityT is the associated type / generic type name used by protocols like IdentityResolver and Signer.
+// - IdentityKind is the enum that's used by IdentityResolverConfiguration to return correct kind of identity resolver
+//   for the given auth scheme. E.g., SigV4AuthScheme has idKind field as .aws. And identityResolver method in SigV4AuthScheme
+//   returns an identity resolver that returns identity of type .aws. 
+public enum IdentityKind: CaseIterable {
+    case aws
+}
```

Comment:
The `IdentityKind` enum is used by the auth schemes to tell the identity resolver config instance what identity type they accept (e.g., `SigV4AuthScheme` accepts `.aws` type), which then returns an identity resolver that returns that identity type if it finds one from identity resolvers configured on the client. 

To convey this information via type information, it would mean the default identity resolver configuration in smithy-swift would have to check the given type value and have a dependency on aws-sdk-swift, which will be a circular dependency. Regarding how this info is handled, SRA writes:
```
Retrieve an identity resolver for the provided identity type .
For Java , we use runtime type information to resolve the specific identity type , 
but other SDKs may use a string ( uniquely identifying each type of identity resolver ) 
or some other language - idiomatic mechanism for an 
auth scheme to extract the specific type of identity resolver it needs . 
``` 
so I figured adding an enum to be used as identity type info container would be good way to resolve this.

Reply:
Had offline discussion, and the enum has now been removed and replaced with using scheme ID of auth schemes instead.